### PR TITLE
Adds Garroting people with a cable coil.

### PR DESCRIPTION
**Pull request by itseasytosee**
### Video here! -->  https://streamable.com/mr5sm9

This PR adds the ability to Garrote People with cable coil and other items from behind.
How do you garrote someone? It's simple!

Firstly, get a length of cable coil at least five units long. (For a tribal alternative try some sinew!)
Secondly, get behind your victim. (Be sure that there are no meddlers nearby)
Thirdly, right-click on your victim with your weapon to get an immediate aggressive grab.

Additionally, your victim will suffer some lose_breath and maybe some damage depending on your tool. After your target is grabbed, you can right-click on them again to upgrade your grab at an accelerated rate. This technique is obviously only practical against unsuspecting victims.

Also adds a fiber wire to the stealth tactical kit.

## Why It's Good For The Game
This adds a new element to improvised combat. Grab based combat isn't used often due to the impracticability of completing two do_afters while your opponent spams their movement keys to break out almost instantly. This, of course, is not an active combat move. It instead can be utilized as a combat opener on an unsuspecting target or as a way to catch retreating enemies.

"_If a man can hit harder with a toolbox, then that same man can grab harder with a cable_" - Thomas "Piano Man" Garrote

## Changelog
:cl: itseasytosee
add: you can now garrote people from behind by right-clicking them with cable coil.
/:cl:

### CODE REVIEW

**Comment by oklews** [comment]
Based? Yes. Extremely fucking busted levels of overpowered? Yes

Aggro grab table slam, laugh at enemy.

**Comment by oklews** [comment]
also, shouldn't you have to target head for this?

**Comment by MrDoomBringer** [comment]
> get an immediate aggressive grab

hahahhah ooohohohohohoh

**Comment by oklews** [comment]
Make it a traitor only ability, and I'm sold on it, otherwise you're just baiting assistants to steal sec gear by just walking behind a sec officer.

File: code/datums/components/garrote.dm [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	if(_damage)
+		damage = _damage
+	target = targ
+	weapon = wpn
+
+	if(HAS_TRAIT(attacker, TRAIT_PACIFISM))
+		to_chat(attacker, "<span class='warning'>You couldn't bring yourself to strangle someone...</span>")
+		return
+	
+	if(!targ.density && !iscarbon(targ))
```
Comment (MrDoomBringer) [inline code/datums/components/garrote.dm:29]:
```suggestion
	if(!targ.density))
```
is there a reason you repeated the iscarbon check here?
also why not move this up to line 14, with the other check?

File: code/game/objects/items/weaponry.dm [hunk 2]
```diff
@@ -842,3 +842,46 @@ for further reading, please see: https://github.com/tgstation/tgstation/pull/301
...
+
+/obj/item/garrote_wire/proc/on_unwield(obj/item/source, mob/user)
+	SIGNAL_HANDLER
+
+	wielded = FALSE
+
+/obj/item/garrote_wire/update_icon_state()
+	icon_state = "fiber_wire1"
+
+/obj/item/fireaxe/suicide_act(mob/user)
```
Comment (MrDoomBringer) [inline code/game/objects/items/weaponry.dm:885]:
```suggestion
/obj/item/garrote_wire/suicide_act(mob/user)
```

**Review by Mothblocks — CHANGES_REQUESTED** [review]
> Grab based combat isn't used often due to the impracticability of completing two do_afters while your opponent spams their movement keys to break out almost instantly. 

thats what its balanced around dumby

File: code/game/objects/items/stacks/sheets/leather.dm [hunk 3]
```diff
@@ -206,6 +206,12 @@ GLOBAL_LIST_INIT(leather_recipes, list ( \
 	singular_name = "wolf sinew"
 	merge_type = /obj/item/stack/sheet/sinew/wolf
 
+/obj/item/stack/sinew/attack_secondary(mob/living/victim, mob/living/user, params)
+	if (user == victim)
+		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+	user.AddComponent(/datum/component/garrote, victim, src, 0.5)
```
Comment (MrDoomBringer) [inline code/game/objects/items/stacks/sheets/leather.dm:212]: are components supposed to be used in this way? Seems weird to constantly stack on components that never get removed if you garrote someone multiple times

**Comment by necromanceranne** [comment]
You know what would be really stellar. Doing me the great service of _killing the resting check in the resist proc and replacing it with a prone check, because I'm a dumbass that thought that was analogous to prone, because I thought PKP would have checked that when HE added it._

Then you'd have a much easier time grabbing people and not have to readd quickgrabs to do it, because then you can just have the garrottes impact resist chance instead of the do_afters.

File: code/datums/components/garrote.dm [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+		attacker.setGrabState(GRAB_AGGRESSIVE)
+		log_combat(attacker, targ, "grabbed", addition="aggressively")
+		playsound(attacker.loc, 'sound/weapons/draw_bow.ogg', 70, frequency = 20000)
+
+		attacker.visible_message("<span class= 'danger'> [attacker] wraps the [wpn] around [targ]'s neck restraining them!", \
+		"<span class= 'danger'>You wrap the [wpn] around [targ]'s neck from behind them! </span>", ignored_mobs = targ)
+		to_chat(targ, "<span class='userdanger'>[attacker] restrains you by wraping the [wpn] around your neck from behind!</span>")
+		targ.dir = attacker.dir //keep them facing away from us
+		return
+	if(targ.pulledby == attacker) //tighten our grip if we are already grabing them
```
Comment (MrDoomBringer) [inline code/datums/components/garrote.dm:48]: ???
so with cable coils you can passivegrab someone then get a free doublespeed aggressive grab on them whenever you want?

File: code/modules/power/cable.dm [hunk 5]
```diff
@@ -537,6 +537,11 @@ GLOBAL_LIST_INIT(wire_node_generating_types, typecacheof(list(/obj/structure/gri
 		amount += extra
 	update_icon()
 
+/obj/item/stack/cable_coil/attack_secondary(mob/living/victim, mob/living/user, params)
+	if (user == victim || amount < 5)
+		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
```
Comment (MrDoomBringer) [inline code/modules/power/cable.dm:542]: some useful feedback in regards to the amount thing would be useful

**Review by MrDoomBringer — CHANGES_REQUESTED** [review]
probably a lot more that i missed

File: code/game/objects/items/stacks/sheets/leather.dm (hunk 3 above)
Comment (necromanceranne) [reply code/game/objects/items/stacks/sheets/leather.dm:212]: A good example might be the currently fixed gunpointing component for something similar to this.

**Comment by Jared-Fogle** [comment]
:-1:

**Comment by itseasytosee** [comment]
Not even 24 hours? Alright, whatever man.

**Comment by MrDoomBringer** [comment]
the video was funny though i liked it

**Comment by Potato-Masher** [comment]
Yet another cool combat option garroted and dragged away.

**Comment by itseasytosee** [comment]
For the record, I have very little sense of game balance. Probably the right call that this is not suited for the current combat system.
This doesn't make garroting people any less badass but hey what can you do.

**Pull request closed without merging**